enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,210 @@
+
+```yaml
+apiVersion: metering.openshift.io/v1
+kind: PostgresTable
+metadata:
+  name: reportdatasource-metering-node-allocatable-cpu-cores
+spec:
+  columns:
+  - name: amount
+    type: float8
+  - name: timestamp
+    type: timestamptz
+  - name: timeprecision
+    type: float8
+  - name: labels
+    type: jsonb
+  databaseName: metering
+  tableName: datasource_metering_node_allocatable_cpu_cores
+  managePartitions: false
+  partitionedBy:
+  - name: dt
+    type: string
+```
+
+Using a Postgresql go-client, the following implementation could be used to insert Prometheus metrics into an existing Postgresql table:
+
+```golang
+	_, err = conn.Exec(context.Background(), fmt.Sprintf("INSERT INTO %s VALUES($1, $2, $3, ($4)::jsonb)", tableName), metric.Amount, metric.Timestamp, metric.StepSize, labels)
+	if err != nil {
+		return err
+    }
+```
+
+#### Updating Default ReportDataSource and ReportQuery Objects
+
+The default, out-of-the-box `ReportDataSource` custom resources promQL queries will need to be updated in order to reference the recording rules that will be defined in the metering rule group. In order to do this, the metering-ansible-operator bundle will need to translate the existing set of promQL queries as a list of recording rules defined in the specification of a `PrometheusRule` custom resource. Support for [Operator bundles can include Prometheus-related custom resource objects](https://github.com/operator-framework/operator-lifecycle-manager/pull/1253) was added in the 0.14.1 OLM release.
+
+Internally, the reporting-operator needs to avoid creating `HiveTable` and `PrestoTable` custom resources and instead favor the creation of a `PostgresTable` custom resource when reconciling `ReportDataSource` and `Report` objects that don't have a database table created as of yet. On a similar note, the proper owner references will need to be injected for all `PostgresTable` custom resources to establish a hierarchy. In the case a user deletes a `Report` (or `ReportDataSource`) object, the Kubernetes garbage collector is responsible for cleaning up the correct `PostgresTable` object, and thus requires no further manual intervention from the user.
+
+#### Extending the `StorageLocation` CRD to Support Postgresql
+
+A `StorageLocation` custom resource is responsible for abstracting away details about the underlying metering storage backend, exposing the minimal configuration required to show the end-user where metering data has been persisted. Currently, this object only supports Hive databases and therefore needs to be able to replicate the same functionality we provide with Hive, but with Postgresql interfaces.
+
+When the metering slim installation has been enabled, the default `StorageLocation` custom resource that gets reconciled by the metering-ansible-operator should attempt to make this `CREATE DATABASE metering ...` database call using the Postgres go-client, and update it's status once that database has been created.
```

Comment:
Oh, I think I am getting a clearer picture: Presto/Hadoop/Hive/HDFS aren't being removed in this enhancement, just disfavored. So on slim install they simply won't be part of Metering, but they could be on non-default install. Should probably bring some version of this working up to the Summary section if I am right.

Reply:
Right, you might want to default to this "slim install" version of metering, where Postgres is doing the heavy-lifting, and then extend out to larger clusters by reconciling the Hadoop/Hive/Presto components as we already know that Hive has support for Postgres and Presto has a Postgres connector. That said, I'll have to work that scenario into the somewhere closer to the introduction for more context.